This does not fix the build when using the mold linker with lto enabled.

I have a failure building on i686 with Glibc 2.23, x86_64 with Glibc 2.27, and armv7l with  Glibc 2.27.

This is the end of the i686 build:
```

libtool: link: gcc -shared  -fPIC -DPIC  lib/.libs/libcrypt_la-alg-des-tables.o lib/.libs/libcrypt_la-alg-des.o lib/.libs/libcrypt_la-alg-gost3411-2012-core.o lib/.libs/libcrypt_la-alg-gost3411-2012-hmac.o lib/.libs/libcrypt_la-alg-hmac-sha1.o lib/.libs/libcrypt_la-alg-md4.o lib/.libs/libcrypt_la-alg-md5.o lib/.libs/libcrypt_la-alg-sha1.o lib/.libs/libcrypt_la-alg-sha256.o lib/.libs/libcrypt_la-alg-sha512.o lib/.libs/libcrypt_la-alg-sm3.o lib/.libs/libcrypt_la-alg-sm3-hmac.o lib/.libs/libcrypt_la-alg-yescrypt-common.o lib/.libs/libcrypt_la-alg-yescrypt-opt.o lib/.libs/libcrypt_la-crypt-bcrypt.o lib/.libs/libcrypt_la-crypt-des.o lib/.libs/libcrypt_la-crypt-gensalt-static.o lib/.libs/libcrypt_la-crypt-gost-yescrypt.o lib/.libs/libcrypt_la-crypt-sm3-yescrypt.o lib/.libs/libcrypt_la-crypt-md5.o lib/.libs/libcrypt_la-crypt-nthash.o lib/.libs/libcrypt_la-crypt-pbkdf1-sha1.o lib/.libs/libcrypt_la-crypt-scrypt.o lib/.libs/libcrypt_la-crypt-sha256.o lib/.libs/libcrypt_la-crypt-sha512.o lib/.libs/libcrypt_la-crypt-sm3.
mold: warning: ./libcrypt.map: cannot assign version `GLIBC_2.0` to symbol `encrypt`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `GLIBC_2.0` to symbol `encrypt_r`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `GLIBC_2.0` to symbol `fcrypt`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `GLIBC_2.0` to symbol `setkey`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `GLIBC_2.0` to symbol `setkey_r`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `XCRYPT_2.0` to symbol `crypt_gensalt_r`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `XCRYPT_2.0` to symbol `xcrypt`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `XCRYPT_2.0` to symbol `xcrypt_gensalt`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `XCRYPT_2.0` to symbol `xcrypt_gensalt_r`: symbol not found
mold: warning: ./libcrypt.map: cannot assign version `XCRYPT_2.0` to symbol `xcrypt_r`: symbol not found
In function '_crypt_sha1_process_bytes',
    inlined from '_crypt_sha1_finish_ctx.isra' at lib/alg-sha1.c:265:3:
lib/alg-sha1.c:244:9: error: 'sha1_do_transform' reading 64 bytes from a region of size 1 [-Werror=stringop-overread]
  244 |         sha1_do_transform (ctx->state, (const uint8_t*)buffer + i);
      |         ^
lib/alg-sha1.c:244:9: note: referencing argument 2 of type 'const uint8_t[64]'
lib/alg-sha1.c: In function '_crypt_sha1_finish_ctx.isra':
lib/alg-sha1.c:103:1: note: in a call to function 'sha1_do_transform'
  103 | sha1_do_transform (uint32_t state[5], const uint8_t buffer[64])
      | ^
In function '_crypt_sha1_process_bytes',
    inlined from '_crypt_sha1_finish_ctx.isra' at lib/alg-sha1.c:267:5:
lib/alg-sha1.c:244:9: error: 'sha1_do_transform' reading 64 bytes from a region of size 1 [-Werror=stringop-overread]
  244 |         sha1_do_transform (ctx->state, (const uint8_t*)buffer + i);
      |         ^
lib/alg-sha1.c:244:9: note: referencing argument 2 of type 'const uint8_t[64]'
lib/alg-sha1.c: In function '_crypt_sha1_finish_ctx.isra':
lib/alg-sha1.c:103:1: note: in a call to function 'sha1_do_transform'
  103 | sha1_do_transform (uint32_t state[5], const uint8_t buffer[64])
      | ^
lto1: all warnings being treated as errors
make[2]: *** [/usr/local/tmp/cc8Rm7nL.mk:2: /usr/local/tmp/ccYDvjDJ.ltrans0.ltrans.o] Error 1
make[2]: *** Waiting for unfinished jobs....
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
mold: fatal: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1646: libcrypt.la] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/libxcrypt.20251106134649.dir/libxcrypt-4.5.0'
make: *** [Makefile:1469: all] Error 2
There was a build error.
`CFLAGS="-O3 -pipe -ffat-lto-objects -fPIC -fuse-ld=mold  -flto=auto -flto=auto" CXXFLAGS="-O3 -pipe -ffat-lto-objects -fPIC -fuse-ld=mold  -flto=auto -flto=auto" FCFLAGS="-O3 -pipe -ffat-lto-objects -fPIC -fuse-ld=mold  -flto=auto -flto=auto" FFLAGS="-O3 -pipe -ffat-lto-objects -fPIC -fuse-ld=mold  -flto=auto -flto=auto" LDFLAGS="-flto=auto" CREW_PRELOAD_ENABLE_COMPILE_HACKS="1" CC_LD="mold" CXX_LD="mold" CREW_PRELOAD_NO_MOLD="0" make -j 4` exited with 2
```

My workaround was disabling lto and using `LD_FLAGS=-Wl,--undefined-version`, which fixes the builds on all three architectures.

Here are the build logs from the GitHub Action build at https://github.com/chromebrew/chromebrew/actions/runs/19137680153/job/54693917795:
i686: [11_Run Updater in container.txt](https://github.com/user-attachments/files/23394353/11_Run.Updater.in.container.txt)
armv7l: [11_Run Updater in container.txt](https://github.com/user-attachments/files/23394337/11_Run.Updater.in.container.txt)
x86_64: [11_Run Updater in container.txt](https://github.com/user-attachments/files/23394359/11_Run.Updater.in.container.txt)

--- a/lib/libcrypt.map.in
+++ b/lib/libcrypt.map.in
@@ -21,12 +21,13 @@ crypt_checksalt		XCRYPT_4.3
 crypt_preferred_method	XCRYPT_4.4
 
 # Interfaces for code compatibility with libxcrypt v3.1.1 and earlier.
-# No longer available to new binaries.
-crypt_gensalt_r		-		XCRYPT_2.0
-xcrypt			-		XCRYPT_2.0
-xcrypt_r		-		XCRYPT_2.0
-xcrypt_gensalt		-		XCRYPT_2.0
-xcrypt_gensalt_r	-		XCRYPT_2.0
+# No longer available to new binaries.  Include in version-script, only
+# if one of the compatibility interfaces is enabled.
+crypt_gensalt_r		-		XCRYPT_2.0:alt:glibc:owl:suse:yes
+xcrypt			-		XCRYPT_2.0:alt:glibc:owl:suse:yes
+xcrypt_r		-		XCRYPT_2.0:alt:glibc:owl:suse:yes
+xcrypt_gensalt		-		XCRYPT_2.0:alt:glibc:owl:suse:yes
+xcrypt_gensalt_r	-		XCRYPT_2.0:alt:glibc:owl:suse:yes
 
 # Deprecated interfaces, POSIX and otherwise; also present in GNU libc
 # since 2.0

